### PR TITLE
Add sniff to ensure spaces around binary operators

### PR DESCRIPTION
**Pull request by carusogabriel**
This is a more generic sniff to the problem that we're trying to solve with #123.

### CODE REVIEW

File: tests/fixed/not_spacing.php [hunk 1]
```diff
@@ -11,7 +11,7 @@
 } elseif (! $test === 0) {
     echo 0;
 } else {
-    echo -1;
+    echo - 1;
```
Comment (morozov) [inline tests/fixed/not_spacing.php]: This looks like a bug. It's a unary minus, should not be affected by the rule.

File: tests/fixed/semicolon_spacing.php [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 declare(strict_types=1);
 
-echo -1;
+echo - 1;
```
Comment (morozov) [inline tests/fixed/semicolon_spacing.php]: This looks like a bug. It's a unary minus, should not be affected by the rule.

File: tests/fixed/not_spacing.php (hunk 1 above)
Comment (carusogabriel) [reply tests/fixed/not_spacing.php]: Please see https://github.com/squizlabs/PHP_CodeSniffer/issues/429.

**Comment by carusogabriel** [comment]
An update on this one: we're still waiting for https://github.com/squizlabs/PHP_CodeSniffer/pull/2456 which will fix the issue raised by https://github.com/doctrine/coding-standard/pull/126#discussion_r286530502.

The solutions presented in https://github.com/squizlabs/PHP_CodeSniffer/issues/2425, such as `PSR12.Operators.OperatorSpacing` does not help us :(

**Comment by Ocramius** [comment]
Deferred to `8.0.0`

**Comment by carusogabriel** [comment]
This was solved via #148.

**Pull request closed without merging**